tests/data_context/test_data_context_datasources.py

```diff
@@ -58,34 +59,146 @@ def test_data_context_instantiates_inline_store_backend_with_filesystem_config(
     assert isinstance(context._datasource_store.store_backend, InlineStoreBackend)
 
 
+@pytest.mark.parametrize(
+    "data_context_fixture_name",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param(
+            "in_memory_runtime_context",
+            id="BaseDataContext",
+        ),
+        pytest.param(
+            "cloud_data_context_in_cloud_mode_with_datasource_pandas_engine",
+            id="DataContext",
+        ),
+    ],
+)
 @pytest.mark.unit
-def test_add_datasource_updates_cache(
-    in_memory_runtime_context: BaseDataContext,
+def test_get_datasource_retrieves_from_cache(
+    data_context_fixture_name: str,
+    request,
 ) -> None:
-    context = in_memory_runtime_context
+    """
+    What does this test and why?
+
+    For both persistence-enabled and disabled contexts, we should always be looking at the
+    cache for object retrieval.
+    """
+    context = request.getfixturevalue(data_context_fixture_name)
+
+    name = context.list_datasources()[0]["name"]
+
+    # If the value is in the cache, no store methods should be invoked
+    with mock.patch(
+        "great_expectations.data_context.store.DatasourceStore.get"
+    ) as mock_get:
+        context.get_datasource(name)
+
+    assert not mock_get.called
+
+
+@pytest.mark.parametrize(
+    "data_context_fixture_name",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param(
+            "in_memory_runtime_context",
+            id="BaseDataContext",
+        ),
+        pytest.param(
+            "cloud_data_context_in_cloud_mode_with_datasource_pandas_engine",
+            id="DataContext",
+        ),
+    ],
+)
+@pytest.mark.unit
+def test_get_datasource_cache_miss(
+    data_context_fixture_name: str,
+    request,
+) -> None:
+    """
+    What does this test and why?
+
+    For persistence-enabled contexts, we should leverage the underlying store in the case
+    of a cache miss.
+
+    For persistence-disabled contexts, we only look at the cache.
```

Comment:
Is this true? I think we still look at the store but it could be an in-memory store

Reply:
nope you are correct! Updated this docstring